core/src/main/kotlin/io/github/serpro69/kfaker/provider/misc/RandomClassProvider.kt

```diff
@@ -105,8 +105,12 @@ class RandomClassProvider {
             config.constructorParamSize == -1
             && config.constructorFilterStrategy == ConstructorFilterStrategy.NO_ARGS
         ) {
-            randomPrimitiveOrNull() as T?
-                ?: constructors.firstOrNull { it.parameters.isEmpty() && it.visibility == KVisibility.PUBLIC }?.call()
+            randomPrimitiveOrNull() as T? ?: try {
+                constructors.firstOrNull { it.parameters.isEmpty() && it.visibility == KVisibility.PUBLIC }?.call()
+            } catch (e: Exception) {
+                e.printStackTrace()
+                throw InstantiationException("Failed to instantiate $this")
```

Comment:
Sorry, don't have an IDE handy but try this:

```
throw InstantiationException("Failed to instantiate $this")
    .initCause(e) as InstantiationException
```

Reply:
Hmm... weird. I've used this several times in Java. I'll try to create a working example tomorrow.

Reply:
This works for me. All messages are there and the stack trace on the console looks complete as well.

```
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import kotlin.reflect.KClass

class FakerTest {
    @Test
    fun `chain exceptions`() {
        val actual = assertThrows<InstantiationException> {
            Number::class.randomClassInstance()
        }

        assertEquals("Failed to instantiate class kotlin.Number", actual.message)
        assertEquals(InstantiationException::class.java, actual.cause?.javaClass)
        assertEquals(null, actual.cause?.message)
        throw actual
    }

    private fun <T : Any> KClass<T>.randomClassInstance(): T {
        val constructor = constructors.firstOrNull()
            ?: error("No constructors found")

        try {
            return constructor.call()
        } catch (e: Exception) {
            throw InstantiationException("Failed to instantiate $this")
                .initCause(e)
        }
    }
}
```

Reply:
right, I was testing it from a unit test, where for some reason I'm not getting proper output (maybe because the unit test also wraps exception somehow)
but yeah, it seems to be working fine, so I'll change this as well - the exception trace definitely looks better this way
thanks for the tip :)

Reply:
Added in #185